feat: add `stats/incr/nanmax`

jalajk3004 opened the pull request:
Resolves #5550

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package nanmax, which works like max in which it calculates max incrementally ignoring NaN values

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5550

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

jalajk3004 committed:
edit

jalajk3004 committed:
edit

jalajk3004 committed:
Merge branch 'develop' of https://github.com/jalajk3004/stdlib into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
added nanmax

jalajk3004 committed:
lint error

jalajk3004 committed:
verify

jalajk3004 committed:
check

jalajk3004 committed:
ident

jalajk3004 committed:
reviewed

jalajk3004 committed:
constructor removed

jalajk3004 committed:
lint

jalajk3004 commented:
@kgryte, Removed all the linting errors, Please review this PR once again.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into nanmax

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into nanmax

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmax/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@stdlib/stats/incr/nanmax",
+  "version": "0.0.0",
+  "description": "Compute a maximum value incrementally, while ignoring `NaN` values.",
```
No backticks around `NaN` in `package.json` descriptions.

```suggestion
  "description": "Compute a maximum value incrementally, while ignoring NaN values.",
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmax/docs/types/index.d.ts:
```diff
@@ -0,0 +1,66 @@
+*
+* ## Notes
+*
+* -   If provided `NaN` or a value which, when used in computations, results in `NaN`, the accumulated value is `NaN` for all future invocations.
```
This note is copied from `incrmax` and contradicts the NaN-ignoring behavior of this package.

```suggestion
* -   If provided `NaN`, the value is ignored and the accumulated value remains unchanged.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmax/examples/index.js:
```diff
@@ -0,0 +1,43 @@
+		v = randu() * 100.0;
+	}
+	max = accumulator( v );
+	console.log( '%d\t%d', v.toFixed( 4 ), ( max === null ) ? NaN : max.toFixed( 4 ) );
```
`.toFixed()` returns a string, so `%d` will coerce it back to a number and strip trailing zeros. Use `%s` instead.

```suggestion
	console.log( '%s\t%s', v.toFixed( 4 ), ( max === null ) ? 'null' : max.toFixed( 4 ) );
```

Planeshifter committed:
fix: address review feedback in stats/incr/nanmax

- Remove incorrect NaN-propagation note from index.d.ts (contradicted the NaN-ignoring behavior)
- Drop backticks from package.json description and align README/repl wording
- Remove inappropriate overflow note (a maximum does not accumulate/sum)
- Fix examples output to use %s and guard against a null accumulated value
- Remove a stray tab in a test title; add an all-NaN returns-null test
- Bump copyright year to 2026

Planeshifter reviewed (approved):
Let's land it!
